iptables/utils_for_test.go

```diff
@@ -336,7 +326,8 @@ func (d *restoreCmd) Run() error {
 					chain[i] = chain[i-1]
 				}
 				chain[0] = rest
-				d.Dataplane.ChainMods.Add(chainMod{name: chainName, ruleNum: 1})
+				ruleNum := d.Dataplane.ChainMods.Len() + 1
```

Comment:
I think this is a bit of a hack; it assumes that the ChainMods set already contains the expected set of data so that we'll get the correct output.  I prefer to avoid that kind of hack in a test since, if the assumption fails, it's usually hard to track down (because you have to know that there's a short-cut being taken in order to figure out why it's giving the wrong result).